modules/core/04-channel/v2/types/packet.go

```diff
@@ -2,10 +2,62 @@ package types
 
 import (
 	"crypto/sha256"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	host "github.com/cosmos/ibc-go/v9/modules/core/24-host"
 )
 
+func (p Packet) ValidateBasic() error {
+	if len(p.Data) != 1 {
+		return errorsmod.Wrapf(ErrInvalidPacket, "packet data length must be 1, got %d", len(p.Data))
+	}
```

Comment:
I think we want this to be the final state for validate basic for the initial release, but in order for mock applications to be testable with multiple packet datas (which I still think is worthwhile doing ), maybe we can leave this check commented out and create an issue to revert it. (or we could just fail on == 0 but create an issue to change)